transform/flatten_mangler.go

```diff
@@ -110,8 +117,13 @@ func (f *FlattenMangler) flattenStruct(fieldPrefix, tagPrefix []string, sf refle
 			flattenedNames = append(fieldPrefix[:len(fieldPrefix):len(fieldPrefix)], nestedsf.Name)
 		}
 
+		// Need a separate flattenPath slice for the field path instead of just
+		// using the fieldPrefix one because we need to add the names of the
+		// embedded fields to the slice so we can iterate through and get the original field
+		flattenedPath := append(fieldPath, nestedsf.Name)
```

Comment:
I think we need to override the capacity here like we do on line 117, so it allocates a new backing array and copies.

Reply:
Yes we do, fixed.